### DISCUSSION TITLE
Modelling time series data

### DISCUSSION THREAD

**Discussion by coedekoven** [inlabru-org/inlabru#161] [General]
Hi Finn,

I am trying to modelling the time series data using inlabru; more specifically I have start times of cetacean vocalizations obtained from single hydrophone recordings as well as start times of military sonar pings from the same recording device. I am trying to identify potential effects of military sonar on acoustic detections of cetacean vocalizations. As I have the start times of both types of detections, I think it would be beneficial to model this process using lgcp() (rather than bru() - which I believe is the equivalent of glm()). I have been working through your workshop materials and practicals, however, two quite fundamental questions remain:

1. what would be a suitable correlation structure for a time series? Would you go with matern where distances are measured in seconds or minutes (whatever the time unit will be)? Do you have to worry about that the flow of time is in one direction, i.e. future events don't have an effect on current or past events?

2. how do you set up a covariate which is essentially a point process itself for a time series? Is it using the fields function?

Looking forward to hearing from you. Cheers, Cornelia

**Comment by finnlindgren** [comment]
`lgcp()` is just a simple wrapper function for single-likelihood models; It essentially does `bru(components, like(formula, family="cp", ...))` ("cp" for Cox process), so you can just as well use `bru()` directly; makes it easier if you want to expand to multilikelihood models (like include marks for the points).

For a time-stationary Poisson point process, the direction of time doesn't matter. It's only when things like self-excitation (Hawkes processes) come in where that makes a difference (we're working on getting a Hawkes processes likelihood implementation aimed at earthquake modelling into the package; current research project).
So I'd a general Matern spde, preferably with `inla.mesh.1d(..., degree=2)` so that the function is more smooth than for `degree=1`.

For the covariate question I'd need to see more details of what model structure you're referring to. If the response is a point process, you need to have the covariates available at every time, regardless of whether there is a point there or not, so I'm not sure what you mean by "a covariate which is essentially a point process itself".

**Comment by finnlindgren** [comment]
Note: I moved this to a Discussion item, since it's not a bug. :)

**Reply by finnlindgren** [reply]
I think I see what you meant now, but what to do depends on what type of model structure you're looking for. I see at least two possibilities:
1. Having the general temporal intensity of the sonar activity as a covariate for the vocalisation process. This can be done with a joint model with two point process likelihoods.
2. Enforcing temporal causality from the sonar points to the vocalisation process.  This would be more similar to a Hawkes process, except that it's not self-exciting; that makes the implementation a bit easier, but still depends on the model details.

I suspect that the mechanism isn't necessarily increasing the vocalisations, but rather suppressing them?  In that case it might make sense to use approach 2 above, with an intensity model
```
lambda(t) = \exp[Intercept + variability(t) + \beta \sum_{j;s_j} kernel(t-s_j; \theta)]
```
where lambda(t) is the intensity for the vocalisation point process, and the kernel(t-s_j; \theta) function is zero for t < s_j, where s_j are the sonar activity time points. The variability(t) would be there to capture non-sonar related activity changes (via a Matern process). Variations of the above would also be possible.
I've never worked on this type of data, so this might not be what you're looking for; let me know.

**Reply by coedekoven** [reply]
Thank you very much, Finn. This is very useful. I will give this a go (your second approach).
